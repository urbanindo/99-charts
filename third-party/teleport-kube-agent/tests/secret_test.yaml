suite: Secret
templates:
  - secret.yaml
tests:
  - it: does not generate a secret when join method is token and neither authToken nor joinParams.tokenName are provided
    asserts:
      - hasDocuments:
          count: 0

  - it: generates a secret when authToken is provided
    set:
      authToken: sample-auth-token-dont-use-this
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Secret
      - equal:
          path: metadata.name
          value: teleport-kube-agent-join-token
      - matchSnapshot: {}

  - it: generates a secret when joinParams.tokenName is provided
    set:
      joinParams:
        tokenName: sample-auth-token-dont-use-this
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Secret
      - equal:
          path: metadata.name
          value: teleport-kube-agent-join-token
      - matchSnapshot: {}

  - it: generates a secret with a custom name when authToken and secretName are provided
    set:
      authToken: sample-auth-token-dont-use-this
      secretName: some-other-secret-name
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Secret
      - equal:
          path: metadata.name
          value: some-other-secret-name
      - matchSnapshot: {}

  - it: sets Secret labels when specified
    values:
      - ../.lint/extra-labels.yaml
    asserts:
      - equal:
          path: metadata.labels.app\.kubernetes\.io/name
          value: teleport-kube-agent
      - equal:
          path: metadata.labels.resource
          value: secret
      - matchSnapshot: {}
