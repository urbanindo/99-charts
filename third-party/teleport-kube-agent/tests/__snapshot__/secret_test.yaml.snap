generates a secret when authToken is provided:
  1: |
    apiVersion: v1
    kind: Secret
    metadata:
      name: teleport-kube-agent-join-token
      namespace: NAMESPACE
    stringData:
      auth-token: |
        sample-auth-token-dont-use-this
    type: Opaque
generates a secret when joinParams.tokenName is provided:
  1: |
    apiVersion: v1
    kind: Secret
    metadata:
      name: teleport-kube-agent-join-token
      namespace: NAMESPACE
    stringData:
      auth-token: |
        sample-auth-token-dont-use-this
    type: Opaque
generates a secret with a custom name when authToken and secretName are provided:
  1: |
    apiVersion: v1
    kind: Secret
    metadata:
      name: some-other-secret-name
      namespace: NAMESPACE
    stringData:
      auth-token: |
        sample-auth-token-dont-use-this
    type: Opaque
sets Secret labels when specified:
  1: |
    apiVersion: v1
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/name: teleport-kube-agent
        resource: secret
      name: teleport-kube-agent-join-token
      namespace: NAMESPACE
    stringData:
      auth-token: |
        auth-token
    type: Opaque
